### DISCUSSION TITLE
Heater relay based energy consumption calculations - is this imprecise?

### DISCUSSION THREAD

**Discussion by AlvinKleine** [veista/nilan#30] [Q&A] [answered]
With following config Relay's "on time" can be calculated for the day - and with that it's easy to sum total consumption of central heating (same for all relays, sum 'em up and multiply by 3kw).

  - platform: history_stats
    name: Time Nilan Relay 1 on
    entity_id: binary_sensor.nilan_heater_relay_1
    state: 'on'
    type: time
    start: '{{ as_timestamp( now().replace(hour=0).replace(minute=0).replace(second=0) ) }}'
    end: '{{ now() }}'
  
...but at least my relay's swap on/off state very frequently, like every minute or two. If integration scan time is 10s can this lead to imprecise outcome? Any thoughts?

**Comment by veista** [comment, marked as answer]
Well, i have installed a separate energy meter on my unit. You can shorten the scan time if it is too long.

**Reply by jonasman** [reply]
which meter btw?

**Reply by veista** [reply]
I have a Quibino Z-Wave 1-phase on this device, but Shelly meters are good also.

**Comment by jonasman** [comment]
The real power is a bit less than 3kW. Make it around 2.85kW.
I measured it with a shelly 3EM pro

**Reply by veista** [reply]
The power consumption depends on mains voltage and water temperature, so it is a coarse approximation.

**Reply by jonasman** [reply]
I tested 1, 2 and 3 relays. It was consistent. 
Voltage here is 230V with a 3kW load. The house was built in 2024, so should be good with the power

**Reply by veista** [reply]
Good for you!
